src/main/ruby/core/string.rb

```diff
@@ -643,21 +643,32 @@ def upto(stop, exclusive=false)
     stop = StringValue(stop)
 
     if stop.size == 1 && size == 1
+      enc = Encoding.compatible?(self.encoding, stop.encoding)
```

Comment:
What if it returns nil though? Should this be be the Rubinius::Type.compatible_encoding ?
```
> "a".upto("b".encode("UTF-16BE")).to_a
Encoding::CompatibilityError: incompatible character encodings: UTF-8 and UTF-16BE
from (pry):6:in `upto'
```

Reply:
Good catch.